menutoplinks.class.php

```diff
@@ -57,17 +57,19 @@ public static function getLinkLang($id_linksmenutop, $id_shop)
         $link = array();
         $label = array();
         $new_window = false;
+        $new_popup = false;
 
         foreach ($ret as $line) {
             $link[$line['id_lang']] = Tools::safeOutput($line['link']);
             $label[$line['id_lang']] = Tools::safeOutput($line['label']);
             $new_window = (bool)$line['new_window'];
+            $new_popup = (bool)$line['new_popup'];
         }
 
-        return array('link' => $link, 'label' => $label, 'new_window' => $new_window);
+        return array('link' => $link, 'label' => $label, 'new_window' => $new_window, 'new_popup' => $new_popup);
     }
 
-    public static function add($link, $label, $newWindow = 0, $id_shop)
+    public static function add($link, $label, $newWindow = 0, $newPopup = 0, $id_shop)
```

Comment:
Isn't the version change sufficient to avoid breackage ?
Do we know about API consumers ?
Isn't the addition more consistent just after $newWindow ?

thank you


On Fri, Aug 07, 2015 at 02:32:16AM -0700, Gregory Roussac wrote:
> >      }
> >  
> > -    public static function add($link, $label, $newWindow = 0, $id_shop)
> > +    public static function add($link, $label, $newWindow = 0, $newPopup = 0, $id_shop)
> 
> Hi again,
> 
> You need to add new parameters at the end of the functions otherwise youwill break backward comptaibility.
> 
> Regards
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/PrestaShop/blocktopmenu/pull/38/files#r36504136

-- 
GPG id: 0xF41572CEBD4218F4